pyHS100/pyHS100.py

```diff
@@ -283,6 +301,173 @@ def identify(self):
 
         return alias, model, features
 
+    @property
+    def alias(self):
+        """
+        Get current device alias (name)
+        :return: Device name aka alias.
+        :rtype: str
+        """
+        return self._alias
+
+    @alias.setter
+    def alias(self, alias):
+        """
+        Sets the device name aka alias.
+        :param alias: New alias (name)
+        :raises SmartPlugException: on error
+        """
+        self._query_helper("system", "set_dev_alias", {"alias": alias})
+
+    @property
+    def led(self):
+        """
+        Returns the state of the led.
+        :return: True if led is on, False otherwise
+        :rtype: bool
+        """
+        return bool(1 - self.sys_info["led_off"])
+
+    @led.setter
+    def led(self, state):
+        """
+        Sets the state of the led (night mode)
+        :param int state: 1 to set led on, 0 to set led off
+        :raises SmartPlugException: on error
+        """
+        self._query_helper("system", "set_led_off", {"off": 1 - state})
+
+    @property
+    def icon(self):
+        """
+        Returns device icon
+
+        Note: this doesn't seem to work (at least) when not using the cloud service.
+
+        :return: icon and its hash
+        :rtype: dict
+        :raises SmartPlugException: on error
+        """
+        return self._query_helper("system", "get_dev_icon")
+
+    @icon.setter
+    def icon(self, icon):
+        """
+        Content for hash and icon are unknown.
+        :param str icon: Icon path(?)
+        :raises NotImplementedError: when not implemented
+        :raises SmartPlugError: on error
+        """
+        raise NotImplementedError("Values for this call are unknown at this point.")
+        # here just for the sake of completeness
+        # self._query_helper("system", "set_dev_icon", {"icon": "", "hash": ""})
+
+    @property
+    def time(self):
+        """
+        Returns current time from the device.
+        :return: datetime for device's time
+        :rtype: datetime.datetime
+        :raises SmartPlugException: on error
+        """
+        response = self._query_helper("time", "get_time")
+        return datetime.datetime(response["year"], response["month"], response["mday"],
```

Comment:
Please, it's already splitted. Changing response to rsp or r doesn't improve readability.